core/hoverfly.go

```diff
@@ -307,6 +307,20 @@ func (hf *Hoverfly) Save(request *models.RequestDetails, response *models.Respon
 		}
 	}
 
+	var headers map[string][]string
+
+	if headersToSave != nil && len(headersToSave) == 0 {
```

Comment:
Yeah, its a bit confusing, CaptureMode, which is calling this function is looking to see if the headers to save is '*' and then just creating an empty list.

I'll move that logic down into here so it makes this code more readable.